video/packet/encode.go

```diff
@@ -0,0 +1,209 @@
+	defer p.endTimeMutex.RUnlock()
+	return p.endTime
+}
+
+func (p *Controller) SetStartTime(s int64) {
+	p.startTime = s
+}
+
+func (p *Controller) SetEndTime(s int64) {
+	p.endTimeMutex.Lock()
+	defer p.endTimeMutex.Unlock()
+	p.endTime = s
+}
+
+func (p *Controller) encode() {
+	var status Status
+	var message string
+	defer func() {
+		if r := recover(); r != nil {
+			status = Error
+			message = fmt.Sprintf("%s", r)
+			p.encodingState.setState(status, message)
+		}
+		if status == Running {
+			status = Error
+			message = "Unknown Error"
+		}
+		p.encodingState.setState(status, message)
+	}()
+
+	status = Running
+	message = "Encoding started..."
+	p.encodingState.setState(status, message)
+
+	dir := os.TempDir()
+	dir = filepath.Join(dir, AppDirName, VideosDirName)
+
+	cachedPath := filepath.Join(dir, CachedFileName)
+	audioPath := filepath.Join(dir, AudioFileName)
+	p.audioPath = audioPath
+
+	_, err := os.Stat(cachedPath)
+	fileExist := !errors.Is(err, os.ErrNotExist)
+
+	if fileExist {
+		file, err := os.Open(cachedPath)
+		if err == nil {
+			b, err := ioutil.ReadAll(file)
+			if err == nil {
+				previousFilePath := string(b)
+				if previousFilePath != p.FilePath() {
+					fileExist = false
+				}
+			}
+		}
+		err = file.Close()
+		if err != nil {
+			fileExist = false
+		} else {
+			message = fmt.Sprintf("File exist, using cached path %s", cachedPath)
+			p.encodingState.setState(status, message)
+		}
+	}
+
+	if !fileExist {
+		err = os.RemoveAll(dir)
+		if err != nil {
+			message = err.Error()
+			status = Error
+			return
+		}
+		err = os.MkdirAll(dir, os.ModePerm)
+		if err != nil {
+			message = err.Error()
+			status = Error
+			return
+		}
+	}
+
+	argSlice := []string{"-i", p.FilePath(), "-show_entries", "format=duration", "-v", "quiet", "-of", "csv=p=0"}
+	command := exec.Command("ffprobe", argSlice...)
+	b, err := command.CombinedOutput()
+	if err != nil {
+		message = err.Error()
+		status = Error
+		return
+	}
+
+	endTimeStr := strings.Split(string(b), "\n")[0]
+	if runtime.GOOS == "windows" {
+		endTimeStr = strings.Split(string(b), "\r\n")[0]
+	}
+	endTimeFl, err := strconv.ParseFloat(endTimeStr, 64)
+	if err != nil {
+		message = err.Error()
+		status = Error
+		return
+	}
+	endTime := int64(endTimeFl)
+
+	p.SetEndTime(endTime)
+
+	command = nil
+	start := time.Now().Unix()
+	for i := int64(0); i <= endTime; i += VideoSize {
+		videoPath := filepath.Join(dir, fmt.Sprintf("%d.mp4", i))
+		currEndTime := VideoSize
+		if endTime-i < VideoSize {
+			currEndTime = endTime - i
+		}
+
+		if !fileExist {
+			// alternate option if -to flag is used instead of t
+			//currEndTime := int(math.Min(float64(p.EndTime()), float64(i + interval)))
+			argSlice := []string{
+				"-ss", fmt.Sprintf("%d", i),
+				"-i", p.FilePath(),
+				"-t", fmt.Sprintf("%d", currEndTime+VideoSizeOffset),
+				"-c:v", "copy",
+				videoPath,
+			}
+			cmd := exec.Command("ffmpeg", argSlice...)
+			_, err := cmd.CombinedOutput()
```

Comment:
```suggestion
			err = cmd.Run()
```
Since we're not using the output, we might as well not ask to collect it.